src/Symfony/Component/Form/Command/FormDebugCommand.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\TableSeparator;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Form\FormRegistryInterface;
+use Symfony\Component\Form\ResolvedFormTypeInterface;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * A console command for retrieving information about form types.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+class FormDebugCommand extends Command
+{
+    private $formRegistry;
+    private $optionsResolver;
+    private $parents = array();
+    private $extensions = array();
+    private $namespaces;
+
+    public function __construct(FormRegistryInterface $formRegistry, array $namespaces = array('Symfony\\Component\\Form\\Extension\\Core\\Type'))
```

Comment:
:+1: i wasnt sure.. as registering form types as a service is not always needed (most core ones are deprecated in fact). So we rely on 2 services (form.type.form / form.type.choice) for the core namespace to be resolved. It sounds a bit fragile :)